src/main/mainWindow.js

```diff
@@ -149,8 +156,10 @@ function createMainWindow(config, options) {
             app.hide();
           });
           mainWindow.setFullScreen(false);
-        } else {
+        } else if (config.minimizeToTray) {
```

Comment:
I think it is integrated in the Settings page: https://github.com/mattermost/desktop/blob/9b5f53a294f05c4b2ad41c7d220e1e7f4064ee02/src/browser/components/SettingsPage.jsx#L307

Why there was such a decision, and whether or not the UX is intended to be the same on all platforms, I cannot decide.

Reply:
seems like for osX is always false, what I meant is that you don't have the option in the settings page for turning it on.  And since there is no option for windows as well, I think it'll show as `undefined` although i haven't check, there might be a default option from the config as well. What I mean is that on win and osx there is no choice for the user.

Reply:
So, would it be possible to simply add this setting to the other platforms, too?

What was the reason to not add this to MacOS or Windows in the first place?

https://github.com/mattermost/desktop/blob/9b5f53a294f05c4b2ad41c7d220e1e7f4064ee02/src/browser/components/SettingsPage.jsx#L847

^ The option should be easy to integrate with MacOS and Windows...

Reply:
@esethna maybe you can comment on this

Reply:
@Willyfrog can you summarize what the user experience change is and what's expected? I'm not sure what's being proposed here. Please ask in the desktop app contributors channel if this is a separate proposed feature enhancement

Reply:
created conversation on [comminity server](https://community-daily.mattermost.com/core/pl/sndhqk8ga7nu9g33ru96np74uy)

Reply:
as per the above conversation, it would be better to restrict the bugfix to linux and file a separate PR if we want to have consistency (I'm 0/5 on this as long as it's optional)

Reply:
@Willyfrog So that means the decision was commonly agreed to move against the UX consistency and do not want to have this integrated with MacOS and Windows (as per removal of the platform check in the Settings jsx file for those platforms)?

Reply:
For this PR, yes. It would be preferred.
But that doesn't mean we don't want to have the option, but it would be a new feature, so it would be better if we do it separately

Reply:
@Willyfrog Who is the former code owner of that decision? Why was it built for Linux only in the first place? What are the pitfalls I would have to test against?

Reply:
Since I wasn't around when this happen, I can only guess that it was because windows and mac default behaviour is different from linux.
On Win, if there is a tray icon it is expected to close the window but have the program running. if there isn't one, then it is expected to quit the app.
On mac, it is expected to keep running wether there is a tray icon or not, and quit on `cmd+q`.

regarding pitfalls, first thing that comes to mind is configuration upgrade and making sure that the app blurs when it is closed/hidden.

Reply:
Regarding Windows: This is the identical behaviour as it is expected on Linux (both GNOME/MATE and KDE).

Regarding MacOS:

- Close Window button must close a window, see this: https://support.apple.com/en-gb/guide/mac-help/mchlp2469/mac
- Restore view feature is missing in general, and Mattermost only has one window; so this is kinda does not apply to Mattermost: https://support.apple.com/en-gb/guide/mac-help/mchl834d18c2/10.15/mac/10.15

Reply:
yes, i think it is working that way, right?

The new feature would be to allow the user to choose it. Like I said, it's not that we don't want it, but have a separate PR so we make sure the bug can be closed wether the feature makes it or not.